thunder/__init__.py

```diff
@@ -581,6 +584,12 @@ def get_computation_and_inputs(*args, **kwargs):
             else:
                 backward_fn = None
 
+            # TODO: using vanilla CUDAGraphExecutor is not safe unless the graph is always static!
+            # (fixme): inspect torch.cuda.make_graph_callables and/or use it instead!
+            # See https://github.com/Lightning-AI/lightning-thunder/issues/433
+            if cd.use_cudagraphs:
+                comp = CUDAGraphExecutor(comp)
```

Comment:
How are network parameters treated? Usually, they can be treated as having static memory and do not require a copy before executing the CUDA graph. Have you verified that it works as expected and no large copy is made before the recorded kernels execution?

Reply:
I did not modify that bit and it is inherited from the original implementation. Which, unless the mask is provided (not the case), marks parameters for not copying.

Reply:
Are nn.Parameters preserved when passed through the prologue trace? I hope they are not unwrapped into tensors.

Reply:
Yes, it is all good there. The mask looks as expected.

Reply:
Thanks a lot for checking!